source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepository.cs

```diff
@@ -14,18 +15,19 @@ namespace Calamari.Deployment.PackageRetention.Repositories
     public class JsonJournalRepository
         : IJournalRepository
     {
+        static readonly SemaphoreSlim JournalSemaphore = new SemaphoreSlim(1, 1);
```

Comment:
We need to use a named semaphore here. `SemaphoreSlim` is only for a single process (including multiple threads), whereas we need this across multiple processes -  Calamari can be run multiple times simultaneously on a single machine.
Use ISemaphoreFactory, because the implementation uses a file-base semaphore when running on mono, and a system one for the rest.

Reply:
Ahh right that makes sense. I added `ISemaphoreFactory` back in, does that look like the right way to dispose of the semaphore?